### PR TITLE
V2xhub release 6.2 fixes

### PR DESCRIPTION
**Pull request by paulbourelly999**
# PR Details
## Description
After integration testing at STOL, discovered some segmentation faults in Qt HTTP client code. Specifically the scope of objects accessed in the lambda function used for signal connections ( essentially Qt call back functions ) was incorrect. QEventLoop object was function scoped and located in a different function than where the connect lambda function was created. This cause segmentation faults for repeated HTTP request.

Additionally SonarScanner discovered several issues in its static code analysis, many of which were addressed in this PR. The PortDrayagePlugin now gets a passing A grade for its static analysis.

Also added fixes for tmxctl CLI calls to obtain correct MySQL credentials. And correct MAX_LOG_LEVEL MACROS to allow for printing log levels other that error.

For Plugin Level Logging now you just need to create a configuration in the manifest JSON called "logLevel" and set its value to a valid log level. The TMX plugin code will then set the plugin log level to the configuration value on every configuration state change.

## Related Issue

#213 
## Motivation and Context

Fixed Logging to allow for V2X-Hub plugin logging to be configurable through V2X-Hub admin web page and allow log statements other than error level to print.
Fixing WebClient Segmentation fault for CARMA-Freight Suntrax testing with V2X-Hub release 6.2. 

Fixing TMX Command Line Interface calls for future use.

## How Has This Been Tested?

Tested in local dev environment with Port Drayage Web Service running and mocking Vehicle communication with UDP socket python script.
## Types of changes

- [ x ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ x] I have added any new packages to the sonar-scanner.properties file
- [ x ] My change requires a change to the documentation.
- [ x ] I have updated the documentation accordingly.
- [ x ] I have read the **CONTRIBUTING** document.
[V2XHUB Contributing Guide](https://github.com/usdot-fhwa-OPS/V2X-Hub/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/v2i-hub/PortDrayagePlugin/src/WebServiceClient.cpp [hunk 1]
```diff
@@ -2,274 +2,267 @@
...
-    polling_frequency = 5;
-    // Initialize API
-    api = new OAIDefaultApi(0);
-    // Setup Server config
-    api->setNewServerForAllOperations(
-        QUrl("http://127.0.0.1:8090"),
-        QString::fromStdString("Unsecured hosting for development"),
-        QMap<QString, OAIServerVariable>()
-    );
+    initialize("127.0.0.1", 8090, false, 5);
```
Comment (snallamothu) [inline src/v2i-hub/PortDrayagePlugin/src/WebServiceClient.cpp:5]: Missing file header.  I noticed that the header information in PortDrayagePlugin.cpp is not consistent with other plugins, please check and let's discuss.

File: src/v2i-hub/PortDrayagePlugin/src/WebServiceClient.h [hunk 2]
```diff
@@ -26,8 +26,10 @@ public slots:
     uint16_t polling_frequency;
     
     // OAIDefaultApi pointer
-    OAIDefaultApi *api;
-
+    std::shared_ptr<OAIDefaultApi> api;
```
Comment (snallamothu) [inline src/v2i-hub/PortDrayagePlugin/src/WebServiceClient.h:29]: Missing file header

File: src/v2i-hub/PortDrayagePlugin/src/WebServiceClient.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (paulbourelly999) [reply src/v2i-hub/PortDrayagePlugin/src/WebServiceClient.cpp:5]: This is just for a default constructor that I don't even think is used anywhere currently. The constructor with parameters is the only one used and it already loads the parameters from the plugin configurations. I can still make the default values constants in the header if that is what you are saying

File: src/v2i-hub/PortDrayagePlugin/src/PortDrayagePlugin.cpp [hunk 3]
```diff
@@ -130,64 +125,36 @@ void PortDrayagePlugin::HandleMobilityOperationMessage(tsm3Message &msg, routeab
...
-					PLOG(logERROR) << "Error occurred during Qt Connection " << std::endl << e.what() << std::endl;
-				}
-			}
-
 		}
 		catch( const ptree_error &e ) {
 			PLOG(logERROR) << "Error parsing Mobility Operation payload: " << e.what() << std::endl;
 		}
+		// Handle actions that require PortDrayage WebService Input
+		if ( pd->operation.compare("PICKUP") == 0 ) {
```
[the remark this answers is not in the document]
Comment (paulbourelly999) [reply src/v2i-hub/PortDrayagePlugin/src/PortDrayagePlugin.cpp:138]: We have a json schema but I can create enums in the header file for this as well

File: src/v2i-hub/PortDrayagePlugin/src/WebServiceClient.cpp (hunk 1 above)
Comment (paulbourelly999) [reply src/v2i-hub/PortDrayagePlugin/src/WebServiceClient.cpp:5]: Not sure what you are referring to

**Pull request merged**